template/php/Dockerfile

```diff
@@ -0,0 +1,52 @@
+# start with the official Composer image and name it
+FROM composer:1.7.1 AS composer
+
+# continue with the official PHP image
+FROM php:7.2-alpine
+
+# copy the Composer PHAR from the Composer image into the PHP image
+COPY --from=composer /usr/bin/composer /usr/bin/composer
+
+# Composer requirements
+ARG COMPOSER_AUTH='{}'
+ENV COMPOSER_AUTH=${COMPOSER_AUTH}
+
+# Add git
+RUN apk add --no-cache git
+
+# Add watchdog
+RUN echo "Pulling watchdog binary from Github."
+RUN curl -sSL https://github.com/openfaas/faas/releases/download/0.9.4/fwatchdog > /usr/bin/fwatchdog
+RUN chmod +x /usr/bin/fwatchdog
+
+# create non-root user
+RUN addgroup -S app && adduser -S -g app app
+RUN mkdir -p /home/app
+
+# Core
+WORKDIR /home/app
+
+# Function
+COPY index.php php-extension.sh composer.* ./
+RUN chmod u+x ./php-extension.sh && ./php-extension.sh
+COPY ./function ./function
+WORKDIR /home/app/function
+
+# Install Dependecies
+RUN [[ -f php-extension.sh ]] && sh ./php-extension.sh || \
+    echo "php-extension.sh does not exist, skip installing PHP extensions"
+RUN [[ -f composer.lock || -f composer.json ]] && composer install --no-dev || \
+    echo "composer.lock does not exist, skip installing composer dependencies"
+
+# Cleanup
+RUN apk del git
```

Comment:
As in,`--no-cache` during the install?

Reply:
See - https://github.com/openfaas/templates/blob/master/template/go/Dockerfile#L7

Reply:
@johnmccabe I can add this in, that isn't a problem but, I'm struggling to find anything online that this does anything? I don't understand a `--no-cache` on a delete since it's a delete? Happy to learn more about how apk works though!

Reply:
Likewise, its in use in the other templates, probably a cut and paste error? It actually ends up pulling down the cache index on delete !!
```
/ # apk del nginx -v --no-cache
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
...
```